### PR TITLE
chore(optimize): add update guide for optimize 3.10

### PR DESCRIPTION
**Pull request by misiekhardcore**
## What is the purpose of the change

_Briefly describe the change you are making, this helps the reviewer by providing an overview._

## Are there related marketing activities

_Are there any marketing activities related to the related feature or component? If so, please briefly describe them._

## When should this change go live?

_Does this change need to be released before a certain date or milestone? If so, when?_

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -938,6 +938,10 @@ module.exports = {
               "Instructions",
               "self-managed/optimize-deployment/migration-update/instructions/"
             ),
+            optimizeLink(
+              "Update notes (3.0 to 3.10)",
```
Comment (andromaqui) [inline sidebars.js:942]: 3.9 to 3.10*

File: optimize/self-managed/optimize-deployment/migration-update/3.9-to-3.10.md [hunk 2]
```diff
@@ -25,3 +25,22 @@ configuration options have therefore been deprecated:
 - ui.header.textColor
 - ui.header.pathToLogoIcon
 - ui.header.backgroundColor
+
+## Elasticsearch
```
Comment (andromaqui) [inline optimize/self-managed/optimize-deployment/migration-update/3.9-to-3.10.md:29]: We should also include that we now support ES8 I think.

**Review by christinaausley — APPROVED** [review]
No adjustments for this one 👍

File: optimize/self-managed/optimize-deployment/migration-update/3.9-to-3.10.md [hunk 3]
```diff
@@ -25,3 +25,22 @@ configuration options have therefore been deprecated:
...
+
+Optimize now supports Elasticsearch `8.x`, but it requires at least Elasticsearch `7.13.0`.
```
Comment (andromaqui) [inline optimize/self-managed/optimize-deployment/migration-update/3.9-to-3.10.md:31]: I would mention the specific versions we support which are "8.5" and "8.6"

**Review by andromaqui — APPROVED** [review]
Approving, just a small change needed. Lmk when you do it and I can reapprove :)

**Comment by misiekhardcore** [comment]
> No adjustments for this one +1 Have all necessary adjustments also been applied to the current `8.1` version of docs where necessary?

I don't know if it's necessary, couldn't find any place where this had to be placed

**Review by andromaqui — APPROVED** [review]
LGTM :)

**Pull request merged**